test/arrayops.jl

```diff
@@ -362,6 +362,14 @@ a = [0,1,2,3,0,1,2,3]
 @test findprev(isodd, [2,4,5,3,9,2,0], 7) == 5
 @test findprev(isodd, [2,4,5,3,9,2,0], 2) == 0
 
+# find with general iterables
+s = "julia"
+@test find(s) == [1,2,3,4,5]
```

Comment:
Ah crap, I didn't realize this was getting the non-null indices; I thought it was getting all indices. I definitely should have added a test for that. This behavior isn't what I had intended.

Would it make sense to add a separate method for strings that just does something like `collect(1:endof(s))`? For things like `Base.UTF8proc.GraphemeIterator` it would still do the string-to-int comparison and return all indices. That would only become an issue if the `NotComparableError` thing happens.

What do you think about that?